### PR TITLE
NEWBODAC-12 add redux-offline

### PR DESCRIPTION
**Pull request by kdondziak**

### CODE REVIEW

File: src/main/webapp/js/components/vaccinee.jsx [hunk 1]
```diff
@@ -1,14 +1,61 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+    };
+  }
+
+  componentDidMount() {
+    this.setState({ loading: true });
+    this.props.fetchVaccinees() //eslint-disable-line
```
Comment (pmuchowski) [inline src/main/webapp/js/components/vaccinee.jsx]: why eslint is disabled?

File: src/main/webapp/js/components/vaccinee.jsx [hunk 2]
```diff
@@ -1,14 +1,61 @@
...
+      .then(() => this.setState({ loading: false }));
+  }
+
+  static getTableColumns() {
```
Comment (pmuchowski) [inline src/main/webapp/js/components/vaccinee.jsx:23]: you will need to write some utils method to create table columns from field config

**Review by pmuchowski — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/components/vaccinee.jsx (hunk 2 above)
Comment (kdondziak) [reply src/main/webapp/js/components/vaccinee.jsx:23]: Should I do it now?

Comment (pmuchowski) [reply src/main/webapp/js/components/vaccinee.jsx:23]: I think it can be added later

File: src/main/webapp/js/actions/types.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const FETCH_VACCINEES = 'fetch_vaccinees';
+export const CREATE_VACCINEES = 'create_vaccinees';
```
Comment (pmuchowski) [inline src/main/webapp/js/actions/types.js]: change it to CREATE_VACCINEE

File: src/main/webapp/js/actions/index.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import apiClient from '../utils/api-client';
+import { FETCH_VACCINEES } from './types';
+
+const BASE_URL = '/api';
+const VACCINEES = `${BASE_URL}/vaccinee`;
+
+export default function () {
+  return function action(dispatch) {
```
Comment (pmuchowski) [inline src/main/webapp/js/actions/index.js:8]: you should use redux-promis here, just return the request

File: src/main/webapp/js/reducers/index.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux';
+import vaccineReducer from './vaccinee-reducer';
+
+const rootReducer = combineReducers({
+  vaccineReducer,
```
Comment (pmuchowski) [inline src/main/webapp/js/reducers/index.js]: I think the state should be called vaccinee, so change it to `vaccinee: vaccineReducer`

File: src/main/webapp/js/reducers/vaccinee-reducer.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import initialState from './initial-state';
+import { FETCH_VACCINEES } from '../actions/types';
+
+export default function (state = initialState, action) {
```
Comment (pmuchowski) [inline src/main/webapp/js/reducers/vaccinee-reducer.js]: use lambda instead of function

File: src/main/webapp/js/components/vaccinee.jsx [hunk 7]
```diff
@@ -1,14 +1,61 @@
...
+    super(props);
+    this.state = {
+      loading: true,
+    };
+  }
+
+  componentDidMount() {
+    this.setState({ loading: true });
+    this.props.fetchVaccinees() //eslint-disable-line
+      .then(() => this.setState({ loading: false }));
```
Comment (pmuchowski) [inline src/main/webapp/js/components/vaccinee.jsx]: then will not work here like you expect

**Review by pmuchowski — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/components/vaccinee.jsx [hunk 8]
```diff
@@ -1,14 +1,60 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+    };
+  }
+
+  componentDidMount() {
+    this.setState({ loading: true });
+    this.props.fetchVaccinees(() => this.setState({ loading: false })) //eslint-disable-line
```
Comment (pmuchowski) [inline src/main/webapp/js/components/vaccinee.jsx]: you can remove the eslint disable

File: src/main/webapp/js/components/vaccinee.jsx [hunk 9]
```diff
@@ -1,14 +1,60 @@
...
+  }
+
+  static getTableColumns() {
+    return [{
+      Header: 'Name',
+      accessor: 'vaccineeId',
```
Comment (pmuchowski) [inline src/main/webapp/js/components/vaccinee.jsx]: change the accessor

**Review by pmuchowski — CHANGES_REQUESTED** [review]

**Review by pmuchowski — APPROVED** [review]

**Pull request merged**